hal/ia32/cpu.c

```diff
@@ -135,62 +135,36 @@ int hal_cpuReschedule(spinlock_t *spinlock, spinlock_ctx_t *scp)
 {
 	int err;
 
-	if (spinlock != NULL) {
-		hal_cpuGetCycles((void *)&spinlock->e);
-
-		/* Calculate maximum and minimum lock time */
-		if ((cycles_t)(spinlock->e - spinlock->b) > spinlock->dmax)
-			spinlock->dmax = spinlock->e - spinlock->b;
-
-		if (spinlock->e - spinlock->b < spinlock->dmin)
-			spinlock->dmin = spinlock->e - spinlock->b;
-	}
-
-	__asm__ volatile (
-		"movl %1, %%eax;"
-		"cmp $0, %%eax;"
-		"je 1f;"
-
-		"movl %3, %%eax;"
-		"pushl %%eax;"
-		"xorl %%eax, %%eax;"
-		"incl %%eax;"
-		"xchgl %2, %%eax;"
-		"jmp 2f;"
-
-		"1:;"
-		"pushf;"
-
-		"2:;"
-		"pushl %%cs;"
-		"cli;"
-		"leal 3f, %%eax;"
-		"pushl %%eax;"
-		"movl $0, %%eax;"
-		"call interrupts_pushContext;"
-
-		"call _interrupts_multilockSet;"
-
-		"leal 0(%%esp), %%eax;"
-		"pushl $0;"
-		"pushl %%eax;"
-		"pushl $0;"
-		"movl %4, %%eax;"
-		"call *%%eax;"
-		"cli;"
-		"addl $12, %%esp;"
-		"popl %%esp;"
-
-		"call _interrupts_multilockClear;"
- 		"pushl %%esp;"
-
-		"jmp interrupts_popContext;"
-
-		"3:;"
-		"movl %%eax, %0"
-	: "=g" (err)
-	: "m" (spinlock), "m" (spinlock->lock), "r" (*scp), "g" (threads_schedule)
-	: "eax", "edx", "esp", "cc", "memory");
+	if (spinlock != NULL)
+		hal_spinlockClear(spinlock, scp);
```

Comment:
spinlock should be cleared only when reschedule is ready - see example here https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/6a2f9bc0bf3eba9829f0a4f271003fa141c0bad3/hal/armv7m/stm32/_init.S#L226 and here https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/6a2f9bc0bf3eba9829f0a4f271003fa141c0bad3/hal/armv7a/_interrupts.S#L129

So at least interrupts should be disabled before clearing spinlock.

But from what I see this implementation wasn't doing that anyway - spinlock clear on beginning, just like on this PR, so no regress here